File: Medium/1838. Frequency of the Most Frequent Element/1838. Frequency of the Most Frequent Element.java
import java.util.Arrays;

public class Solution {
    public int maxFrequency(int[] nums, int k) {
        Arrays.sort(nums); // Sort the array

        int maxFreq = 0;
        int left = 0;
        long sum = 0; // Use long to avoid integer overflow

        for (int right = 0; right < nums.length; right++) {
            sum += nums[right];

            // Check if we need to shrink the window
            while ((right - left + 1) * (long) nums[right] - sum > k) {
                sum -= nums[left];
                left += 1;
            }

            // Update the maximum frequency
            maxFreq = Math.max(maxFreq, right - left + 1);
        }

        return maxFreq;
    }
}

File: Medium/1838. Frequency of the Most Frequent Element/1838. Frequency of the Most Frequent Element.rs
impl Solution {
    pub fn max_frequency(nums: Vec<i32>, k: i32) -> i32 {
        let mut nums = nums;
        nums.sort(); // Sort the array

        let mut max_freq = 0;
        let mut left = 0;
        let mut sum = 0;

        for right in 0..nums.len() {
            sum += nums[right];

            // Check if we need to shrink the window
            while (right - left + 1) as i32 * nums[right] - sum > k {
                sum -= nums[left];
                left += 1;
            }

            // Update the maximum frequency
            max_freq = max_freq.max(right - left + 1);
        }

        max_freq as i32
    }
}
